### PR TITLE
Additional Security Headers

### PR DESCRIPTION
**Pull request by alex-kovoy**
**Purpose**

Sets addition HTTP response headers that instruct web agents not to cache HTTP traffic and apply additional security policies supported by modern browsers.

**Implementation**

* Disabled caching with the following headers.
  * `Cache-Control:  no-cache, no-store, must-revalidate`
  * `Pragma: no-cache`
  * `Expires: 0`
* Only allow same origin iframes.
  * `X-Frame-Options: SAMEORIGIN`
* Detect and prevent reflected cross-site scripting (XSS) attacks in modern browsers.
  * `X-XSS-Protection: 1; mode=block`
* Require all future requests to be HTTPS.
  * `Strict-Transport-Security: max-age=31536000; includeSubDomains`
* Prevent web browsers from using content sniffing to discover a file’s MIME type.
  * `X-Content-Type-Options: nosniff`
* Set Content Security Policy.
  * `Content-Security-Policy: script-src 'self'; style-src 'self' 'unsafe-inline'; object-src 'none'; img-src 'self' data: blob:`

**Related Issues**

Fixes https://github.com/gravitational/teleport/issues/1148

### CODE REVIEW

**Comment by alex-kovoy** [comment]
gravitational-jenkins retest this please

File: lib/httplib/httplib.go [hunk 1]
```diff
@@ -40,6 +41,9 @@ type StdHandlerFunc func(w http.ResponseWriter, r *http.Request) (interface{}, e
 // MakeHandler returns a new httprouter.Handle func from a handler func
 func MakeHandler(fn HandlerFunc) httprouter.Handle {
 	return func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+		// ensure that niether proxies nor browsers cache http traffic
```
Comment (russjones) [inline lib/httplib/httplib.go]: neither

File: lib/httplib/httplib.go [hunk 2]
```diff
@@ -90,13 +94,43 @@ func ConvertResponse(re *roundtrip.Response, err error) (*roundtrip.Response, er
...
+	// Prevent web browsers from using content sniffing to discover a file’s MIME type
+	h.Set("X-Content-Type-Options", "nosniff")
+
+	// Set content policy flags
+	var cspValue = strings.Join([]string{
+		"script-src 'self'",
+		// 'unsafe-inline' needed for reactjs inline styles
+		"style-src 'self' 'unsafe-inline'",
+		"object-src 'none'",
+		"img-src 'self' data: blob:",
```
Comment (russjones) [inline lib/httplib/httplib.go]: Should this header end with `blob:`?

**Review by russjones — APPROVED** [review]

File: lib/httplib/httplib.go (hunk 2 above)
Comment (alex-kovoy) [inline lib/httplib/httplib.go]: yes, this is a valid value

Comment (russjones) [inline lib/httplib/httplib.go]: Okay, cool.

**Comment by alex-kovoy** [comment]
gravitational-jenkins retest this please

**Pull request merged**